.github/workflows/release.yml

```diff
@@ -0,0 +1,117 @@
+name: Publish to crates.io
+
+# on:
+#   push:
+#     tags:
+#       - 'v*'
+
+on:
+  pull_request: # Test trigger
+
+env:
+  RUSTFLAGS: -Dwarnings
+  RUST_BACKTRACE: 1
+  # Change to specific Rust release to pin
+  rust_stable: stable
+  rust_nightly: nightly-2024-07-07
+  # When updating this, also update relevant docs
+  rust_min: '1.81'
+  RELEASE_ROLE_ARN: ${{ secrets.RELEASE_ROLE_ARN }}
+  TOKEN_KEY_NAME: ${{ secrets.TOKEN_KEY_NAME }}
+  AWS_REGION: us-west-2
+  # Test values
+  TEST_DRY_RUN: 'true'
+
+permissions:
+  id-token: write  # This is required for requesting the JWT
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install Rust ${{ env.rust_nightly }}
+        uses: dtolnay/rust-toolchain@stable
+        with:
+            toolchain: ${{ env.rust_stable }}
+
+      - name: Install cargo-nextest
+        uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-nextest
+
+      - uses: Swatinem/rust-cache@v2
+
+      - name: Verify tag matches Cargo.toml version
+        shell: bash
+        working-directory: ./aws-s3-transfer-manager/
+        run: |
+          # Determine the tag - use actual tag or a test tag
+          if [[ "${{ github.event_name }}" == "push" && "${{ github.ref_type }}" == "tag" ]]; then
+            # Extract tag from GitHub ref (e.g. refs/tags/v1.2.3 -> v1.2.3)
+            TAG="${{ github.ref_name }}"
+          else
+            # If not triggered by a tag push, use a test tag for testing
+            TAG="v0.1.0"
+            echo "Using test tag: $TAG"
+          fi
+
+          # Extract version from tag (remove 'v' prefix)
+          TAG_VERSION="${TAG#v}"
+
+          # Extract version from Cargo.toml
+          CARGO_VERSION=$(grep -m 1 '^version = ' Cargo.toml | sed 's/version = "\(.*\)"/\1/')
+
+          echo "Tag version: $TAG_VERSION"
+          echo "Cargo.toml version: $CARGO_VERSION"
+
+          # Verify versions match
+          if [[ "$TAG_VERSION" != "$CARGO_VERSION" ]]; then
+            echo "::error::Version mismatch! Git tag ($TAG_VERSION) does not match Cargo.toml version ($CARGO_VERSION)"
+            exit 1
+          else
+            echo "✅ Version check passed: Git tag matches Cargo.toml version ($TAG_VERSION)"
+          fi
+
+      - uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ env.RELEASE_ROLE_ARN }}
```

Comment:
What account is this pulling from?

Reply:
the rust sdk production account, I agree we can move it to a separate account, but i still want one single place to host the token and rotate it, instead of copy the string around.